docs/PROTOCOL.md

```diff
@@ -0,0 +1,243 @@
+   - Set the channel status to `ACTIVE`
+   - Emit an `Opened` event with the channelId
+
+## 6. Off-Chain State Updates
+
+1. Participants MUST:
+   - Exchange and sign state updates off-chain
+   - Maintain a record of the latest valid state
+   - Use application-specific data in the `state.data` field
+
+2. Each new state MUST:
+   - Contain updated allocations reflecting the new distribution of assets
+   - Be signed by the necessary participants according to application rules
+   - Comply with the validation rules of the channel's adjudicator
+
+## 7. Channel Closure
+
+### 7.1. Cooperative Close
+
+1. To close cooperatively, any participant MAY:
+   - Prepare a final state where `state.data` MUST be set to the magic number `CHANCLOSE` (7879)
+   - Collect signatures from all participants on this final state
+   - Call the `close` function with the channelId, final state, and any required proofs
+
+2. The system MUST:
+   - Verify all participant signatures on the closing stateHash
+   - Verify the state contains the `CHANCLOSE` magic number
+   - Distribute funds according to the final state's allocations
+   - Set the channel status to `FINAL`
+   - Delete the channel and emit a `Closed` event
+
+### 7.2. Challenge-Response Process
+
+1. To initiate a challenge, a participant MAY:
+   - Call the `challenge` function with their latest valid state and required proofs
+
+2. The system MUST:
+   - Verify the submitted state via the adjudicator
+   - If valid, store the state and start the challenge period
+   - Set a challenge expiration timestamp (current time + challenge duration)
+   - Set the channel status to `DISPUTE`
+   - Emit a `Challenged` event with the channelId and expiration time
+
+3. During the challenge period, any participant MAY:
+   - Submit a more recent valid state by calling `challenge` again
+   - If the new state is valid and more recent (as determined by the adjudicator or `IComparable`), the system MUST update the stored state and reset the challenge period
+
+4. After the challenge period expires, any participant MAY call `reclaim` to distribute funds according to the last valid challenged state
+
+### 7.3. Checkpointing
+
+1. Any participant MAY:
+   - Call the `checkpoint` function with a valid state and required proofs
+
+2. The system MUST:
+   - Verify the submitted state via the adjudicator
+   - If valid, store the state without starting a challenge period
+   - Emit a `Checkpointed` event with the channelId
+
+## 8. Reset Protocol
+
+1. To reset a channel (for resizing allocations), any participant MAY:
+   - Call the `reset` function with:
+     - The current channelId
+     - A valid closing state with all signatures
+     - Any required proofs
+     - A new channel configuration
+     - A new initial state with the `CHANOPEN` magic number
+
+2. The system MUST:
+   - Close the existing channel as per the cooperative close process
+   - Create a new channel following the creation process
```

Comment:
"Create channel", does it mean, that configs could be different. Like different participants/challenge or just different allocations?

Reply:
I also thought about it, but on some lines above it is clearly stated that "Sign the *same* funding stateHash". If anything is changed, then it would not be the same stateHash.

Reply:
Yes, During funding process we should save initial state and make sure stateHash in the same when new participant are joining. I will add in the spec